apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -286,3 +292,23 @@ function wpcom_display_global_styles_banner_extra_tooltip() {
 	<?php
 }
 add_action( 'launch_bar_extra_tooltip_toggle_global_styles', 'wpcom_display_global_styles_banner_extra_tooltip' );
+
+/**
+ * Invalidates the cached Global Styles after changing them or when previewing them.
+ */
+function wpcom_invalidate_global_styles_cache() {
+	$transient_name = 'gutenberg_global_styles_' . get_stylesheet();
+
+	if ( wpcom_should_limit_global_styles() && wpcom_is_previewing_global_styles() ) {
+		add_filter( 'pre_transient_' . $transient_name, '__return_null' );
+		add_filter( 'pre_set_transient_' . $transient_name, '__return_false' );
+	}
+
+	add_action(
+		'save_post_wp_global_styles',
+		function () use ( $transient_name ) {
+			delete_transient( $transient_name );
+		}
+	);
```

Comment:
I would really love if we could actually fix the issue in Gutenberg instead of circumventing it. With this code, we aren't preventing the issue from happening to other components/systems that use global styles. Gutenberg should be purging the cache whenever the Global Styles CPT gets updated.

Reply:
Sure, I'll create an equivalent PR for both Core and Gutenberg tomorrow. But as noted above, we would still need this PR since otherwise we would have the cache problems for weeks.

Reply:
> But as noted above, we would still need this PR since otherwise we would have the cache problems for weeks.

I think that we can wait, Gating Global Styles isn't live so technically the cache problem doesn't exist yet. There still needs to be some conversations before we can actually resume working on Gating Global Styles. We won't be launching the feature any time soon, but I'm ok with merging this as long as we add a follow-up task to remove the code once the issue is fixed in Core/Gutenberg.

I really believe that we aren't in a hurry with this and should prioritize fixing the underlying problem rather than making it work for our specific case, but as I said above I don't mind releasing this as long as we take ownership

Reply:
> Gating Global Styles isn't live so technically the cache problem doesn't exist yet

It does exist, since the cache currently prevents all WP.com users from immediately see changes done in the site editor.

Said that, [the preferred approach by Core](https://core.trac.wordpress.org/ticket/56910) seems to cache the styles forever (instead of a minute) and correctly invalidate them. This makes the invalidation harder, since we need to account for many more scenarios like WordPress updates, theme updates, etc. However, this is currently investigated in [this PR that seeks to cache the `gutenberg_get_global_settings` method](https://github.com/WordPress/gutenberg/pull/45372#issuecomment-1296859561), so I think we should wait for that one first and replicate the same strategy afterwards.

Reply:
> Gutenberg should be purging the cache whenever the Global Styles CPT gets updated.

This is the key issue we need to address, in my view. Marin and I talked about https://github.com/WordPress/gutenberg/pull/41201#issuecomment-1303111763, and I'd love this is addressed in addition to migrate away from transients to object cache.

Reply:
I also want to share https://github.com/WordPress/gutenberg/issues/45171 for you to see. I've seen many issues with caches, so I created that to create some clarity as to what needs fixing.